File: Creational/Builder/PHP/products/CarManual.php
<?php

require_once(__DIR__.'/../interfaces/IEngine.php');

/**
 * Some concrete product.
 */

class CarManual
{
	public function doStuff()
	{
		print("Car manual has info about engine ".$this->_engine->getType()." ".$this->_engine->getVolume().".\n");
		print("Car manual has info about {$this->_seats} seats.\n");
		print("Car manual has info about trip computer: ".($this->_tripComputer ? 'Yes' : 'No').".\n");
	}

	/**
	 * @param string $engine
	 */
	public function setEngine(IEngine $engine)
	{
		$this->_engine = $engine;
		return $this;
	}

	/**
	 * @param int $seats
	 */
	public function setSeats($seats)
	{
		$this->_seats = $seats;
		return $this;
	}

	/**
	 * @param boolean $tComputer
	 */
	public function setTripComputer($tComputer = true)
	{
		$this->_tripComputer = $tComputer;
		return $this;
	}

	private $_engine;
	private $_seats;
	private $_tripComputer;
}

File: Creational/Builder/C++/products/CarManual.cpp
#include <iostream>
#include "CarManual.h"
#include "./../interfaces/IEngine.h"

void CarManual::doStuff()
{
    std::cout << "Car manual has engine " << _engine->getType() << " "
        << _engine->getVolume() << std::endl;
    std::cout << "Car manual has " << _seats << " seats" << std::endl;
    std::cout << "Car manual has trip computer: " << (_tripComputer ? "Yes" : "NO") << std::endl;
}

CarManual* CarManual::setEngine(IEngine* engine)
{
    _engine = engine;
    return this;
}

CarManual* CarManual::setSeats(const int seats)
{
    _seats = seats;
    return this;
}

CarManual* CarManual::setTripComputer(const bool tComputer)
{
    _tripComputer = tComputer;
    return this;
}
